From jirislaby@gmail.com  Wed Mar 11 14:00:49 2009
From: Jiri Slaby <jirislaby@gmail.com>
Date: Wed, 11 Mar 2009 21:47:36 +0100
Subject: USB: atm/cxacru, fix lock imbalance
To: gregkh@suse.de
Cc: Jiri Slaby <jirislaby@gmail.com>, Simon Arlott <cxacru@fire.lp0.eu>
Message-ID: <1236804460-1432-1-git-send-email-jirislaby@gmail.com>


We do not hold mutex in one place in cxacru_cm, but unlock it on fail path.
Fix this.

Signed-off-by: Jiri Slaby <jirislaby@gmail.com>
Cc: Simon Arlott <cxacru@fire.lp0.eu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/atm/cxacru.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/usb/atm/cxacru.c
+++ b/drivers/usb/atm/cxacru.c
@@ -485,7 +485,7 @@ static int cxacru_cm(struct cxacru_data 
 			usb_err(instance->usbatm, "requested transfer size too large (%d, %d)\n",
 				wbuflen, rbuflen);
 		ret = -ENOMEM;
-		goto fail;
+		goto err;
 	}
 
 	mutex_lock(&instance->cm_serialize);
@@ -565,6 +565,7 @@ static int cxacru_cm(struct cxacru_data 
 	dbg("cm %#x", cm);
 fail:
 	mutex_unlock(&instance->cm_serialize);
+err:
 	return ret;
 }
 
